indra_gpt/api.py

```diff
@@ -115,3 +117,59 @@ def run_openai_chat(
         print(f"Response:\n---------\n{response}\n---------\n\n")
 
     return reply
+
+def run_openai_chat_batch(prompts, chat_histories, model, max_tokens):
+
+    batch_requests = []
+
+    for i, (prompt, history) in enumerate(zip(prompts, chat_histories)):
+        messages = history + [prompt]
+        batch_requests.append(
+            {"custom_id": f"request-{i}", "method": "POST", "url": "/v1/chat/completions", "body": {"model": model, "messages": messages, "max_tokens": max_tokens}}
+        )
+
+    # Make a temporary directory in the directory one level above the directory of this file
+    tmp_dir_path = os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__))), "tmp")
+    os.makedirs(tmp_dir_path, exist_ok=True)
+    # Write the batch requests to a file
+    batch_input_file_path = os.path.join(tmp_dir_path, "batch_input.jsonl")
```

Comment:
IMO using `pathlib.Path` would make it more clear:
```suggestion
    # Make a temporary directory in the directory one level above the directory of this file
    tmp_dir_path = Path(__file__).resolve().parent.parent / "tmp"
    tmp_dir_path.mkdir(parents=True, exist_ok=True)
    # Write the batch requests to a file
    batch_input_file_path = tmp_dir_path / "batch_input.jsonl"
```